vtex-apps/store-graphql
# Add filtering of documents search with `_where`

#### What is the purpose of this pull request?
- Enable filtering of MasterData v1 documents search
- Fix a bug in the creation of `VtexIdclientAutCookie` header

#### What problem is this solving?
- Returning documents that fit into the filter value
- Bug in `withAuthCookie`

#### How should this be manually tested?
- Execute the query in the link below with all the variables, it should return only one user, in that case, the one that has my email address registered on MasterData
- Remove the `"where"` param on the query variables and refetch the query, it should now return all documents (in the linked store, there are only two documents) 

#### Screenshots or example usage
[GraphiQL](https://documentsearchfilterpr--delivery.myvtex.com/_v/vtex.store-graphql@2.31.7/graphiql/v1?operationName=documents&query=query%20documents(%24acronym%3A%20String%2C%20%24fields%3A%20%5BString%5D%2C%20%24where%3A%20String)%20%7B%0A%20%20documents(acronym%3A%20%24acronym%2C%20fields%3A%20%24fields%2C%20where%3A%20%24where)%20%7B%0A%20%20%20%20id%0A%20%20%20%20fields%20%7B%0A%20%20%20%20%20%20key%0A%20%20%20%20%20%20value%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D&variables=%7B%0A%20%20%22acronym%22%3A%20%22CL%22%2C%0A%20%20%22fields%22%3A%20%5B%0A%20%20%20%20%22email%22%0A%20%20%5D%2C%0A%20%20%22where%22%3A%20%22email%3Ddaniel.carvalho%40penseavanti.com.br%22%0A%7D)

#### Types of changes
